### ISSUE TITLE
Add api to remove param's from nodes

### ISSUE THREAD

**Issue by gmorenz** [setzer22/egui_node_graph#32]
Currently we can add parameters, but not remove them. It would be nice to be able to do both, to be able to do things like have math nodes that take a variable number of inputs, or to be able to grow/shrink the number of ouputs so that there's always another free one (for... reasons).

I've implemented this for output param's in 96dc5b6f672a5f8096d5f85b71ed2234ed835f27 - which depends on #30.

**Comment by setzer22** [comment]
Indeed, that would be quite useful. It's good that more people are using the library so we can test it against a variety of use cases :slightly_smiling_face: I'm happy to accept a PR for this once #30 gets merged
